Userland/Libraries/LibGUI/GlyphMapWidget.cpp

```diff
@@ -52,7 +52,6 @@ GlyphMapWidget::GlyphMapWidget()
 {
     set_focus_policy(FocusPolicy::StrongFocus);
     horizontal_scrollbar().set_visible(false);
-    did_change_font();
```

Comment:
did_change_font() has the effect of doing all this work:

```
    recalculate_content_size();
    vertical_scrollbar().set_step(font().pixel_size_rounded_up() + m_vertical_spacing);
```

Seeing as it calls font() and uses methods on it directly, we must have a font already.

Calling did_change_font() here in the constructor has the effect of setting the content size (to 0, presumably) and setting the vertical scrollbar step to that of the default font per the FontDatabase.

See here in the widget constructor (a few base classes up)

```
Widget::Widget()
    : Core::Object(nullptr)
    , m_background_role(Gfx::ColorRole::Window)
    , m_foreground_role(Gfx::ColorRole::WindowText)
    , m_font(Gfx::FontDatabase::default_font())
```

Do we know that all users set a custom font before displaying the GlyphMapWidget? If not, it seems perfectly reasonable to call this method to set up the internal state for the default system font.

Reply:
A better change to this class might be to explicitly initialize its base class to make the object heirarchy clearer here

```
GlyphMapWidget::GlyphMapWidget()
   : AbstractScrollableWidget()
{
    set_focus_policy(FocusPolicy::StrongFocus);
```

Reply:
I see the issue now. Let me explain why I removed this line:

I assumed that calling `did_change_font()` in the constructor was breaking something due to line numbers being inaccurate (*note to self, never trust line numbers from optimized code*), so I attempted to remove it, which did "solve" the problem but indirectly (again, I'm not even sure why)

The actual issue was being triggered in `paint_event`. It was trying to access `m_original_font`, however it's only initialized if `set_font` is called:
https://github.com/SerenityOS/serenity/blob/e4b71495f5a83817bcd56b75f625076016b41253/Userland/Libraries/LibGUI/GlyphMapWidget.cpp#L167-L169

The fix to this, from my understanding, is to just set the `m_original_font` in the constructor to the `AbstractScrollableWidget`'s font.

```cpp
GlyphMapWidget::GlyphMapWidget()
    : AbstractScrollableWidget()
{
    m_original_font = font().clone();
```